lua/codeium/util.lua

```diff
@@ -35,8 +36,50 @@ function M.get_newline(bufnr)
 	return enums.line_endings[vim.bo[bufnr].fileformat] or "\n"
 end
 
-function M.get_relative_path(bufnr)
-	return vim.fn.fnamemodify(vim.api.nvim_buf_get_name(bufnr), ":")
+local cached_roots = {}
+local function _get_project_root()
+	if config.options.workspace_root.find_root then
+		root = config.options.workspace_root.find_root()
+		if root then
+			return root
+		end
+	end
+
+	if config.options.workspace_root.use_lsp then
+		root = vim.lsp.buf.list_workspace_folders()[0]
+		if root then
+			return root
+		end
+	end
+
+	-- From the CWD, walk up the tree looking for a directory that contains one of the project root files
+	local candidates = config.options.workspace_root.paths
+	local result = vim.fs.find(candidates, {
+		path = cwd,
```

Comment:
@dimfeld where is this cwd variable coming from? I notice that this pr fixed things for mac, but on windows it fails. Seems the failure is because this cwd variable is nil. How does it get populated on mac?

Reply:
Ah sorry, that was missed when I was moving around some code. `M.get_project_root` should pass its `wd` variable to `_get_project_root` for the value of `cwd`.

Reply:
Submitted #248 as a fix